### PR TITLE
Added Auto-Login, Logout, and Exit feature

### PR DESCRIPTION
**Pull request by dixonyant**
- User can login to the game
- User can logout of the game at any point, exiting the game
- User can exit the game, retaining their login credentials
- Game will auto-login the user if the user has not logged out of the game

### CODE REVIEW

File: stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs [hunk 1]
```diff
@@ -10,31 +10,83 @@
...
+    public Button exitButton;
+    public LogoutCanvas logoutCanvas;
+
     private void Awake()
     {
         if (Instance == null)
         {
-            DontDestroyOnLoad(gameObject);
             Instance = this;
+            DontDestroyOnLoad(gameObject);
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs:30]: updated the format of a dontdestroyonload according to a newer template recommendation

File: stc_game/Assets/LogoutCanvas.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class LogoutCanvas : MonoBehaviour
+{
+    public Canvas logoutCanvas;
+    // Start is called before the first frame update
+    void Start()
+    {
+        logoutCanvas = gameObject.GetComponent<Canvas>();
+    }
+
+    // Update is called once per frame
```
Comment (tcape) [inline stc_game/Assets/LogoutCanvas.cs]: Empty Update. Sloppy sloppy. tsk tsk lol

File: stc_game/Assets/Scripts/Authentication/AuthService.cs [hunk 3]
```diff
@@ -26,15 +26,37 @@ public class AuthService {
...
+    public void LoginExistingUser()
+    {
+        AuthRes res = new AuthRes();
+        res.access_token = PlayerPrefs.GetString(GameStrings.LocalStorage.AuthToken);
+        GetAuthUser(res);
+    }
+
+    public bool isLoggedIn()
+    {
+        if (PlayerPrefs.GetString(GameStrings.LocalStorage.AuthToken) != "")
```
Comment (tcape) [inline stc_game/Assets/Scripts/Authentication/AuthService.cs]: could be 1 line even! I don't really care.

**Review by tcape — APPROVED** [review]

File: stc_game/Assets/LogoutCanvas.cs (hunk 2 above)
Comment (dixonyant) [reply stc_game/Assets/LogoutCanvas.cs]: Nice!!

**Pull request merged**